og.module

```diff
@@ -2775,21 +2776,19 @@ function og_get_user_roles($group_type, $gid, $uid = NULL, $include = TRUE, $che
     $uid = $user->uid;
   }
 
-  $account = user_load($uid);
-
   $identifier = implode(':', array($group_type, $gid, $uid, $include));
   if (isset($roles[$identifier])) {
     return $roles[$identifier];
   }
 
-  $is_blocked = og_is_member($group_type, $gid, 'user', $account, array(OG_STATE_BLOCKED));
+  $is_blocked = og_is_member($group_type, $gid, 'user', $uid, array(OG_STATE_BLOCKED));
```

Comment:
This doesn't seem right, the signature of og_is_member still expects an entity yet here you pass an integer.

It might work if og_get_user_roles was called with $uid=NULL but otherwise it'll fail.

Reply:
`og_is_member()` passes the `$entity` parameter to `og_get_entity_groups()`, it doesn't need the parameter for anything else. `og_get_entity_groups()` accepts either a user object or a user ID. In fact, when an object is passed only the ID is used:

```php
if (is_object($entity)) {
    // Get the entity ID.
    list($id) = entity_extract_ids($entity_type, $entity);
  }
```

The docblock of `og_is_member()` could be updated to reflect that.